lib/meta_logger/formatter.ex

```diff
@@ -5,7 +5,7 @@ defprotocol MetaLogger.Formatter do
 
   @fallback_to_any true
 
-  @spec format(struct()) :: String.t() | list()
+  @spec format(struct()) :: no_return()
```

Comment:
Why `no_return` here?

Reply:
Because the "any" function: https://github.com/FindHotel/meta_logger/blob/d41b51fdbff7ef42927a5ffa0669fae2cf3a732d/lib/meta_logger/formatter.ex#L75-L92

Dialyzer does not check the macro, only the public functions. 🍆 

Similar to https://github.com/FindHotel/bofh-api/blob/3bf110e0287b454ccdedfb2ff4499e61615d4d75/apps/api/lib/api_web/default/json_encoder.ex#L11.